src/pudl/transform/eia861.py

```diff
@@ -552,6 +552,10 @@ def _pre_process(df: pd.DataFrame) -> pd.DataFrame:
 
 def _post_process(df: pd.DataFrame) -> pd.DataFrame:
     """Post-processing applied to all EIA-861 dataframes."""
+    # If we're only processing some years of data, we may have entirely empty dataframes
+    # in the extraction phase, in which case the data_maturity field doesn't get added.
+    if "data_maturity" not in df.columns and len(df) == 0:
+        df["data_maturity"] = pd.NA
```

Comment:
nit, because this function is already here, so we might as well use it: putting an extraction step in the transform module is a weird architectural choice

Reply:
I poked at trying to add `data_maturity` in the extract where it's being set for non-empty dataframes, and didn't see an ideal place -- since the code that does that work doesn't get run at all if the dataframe is going to end up empty. Instead I moved it into the `_pre_process()` function in this module, which means it doesn't need to be anywhere else in here. The only other changes now are to accommodate the possibility of having zero rows in division.